app/src/main/java/com/example/androidacademyhw/details/FragmentMoviesDetails.kt

```diff
@@ -74,7 +75,7 @@ class FragmentMoviesDetails : Fragment()  {
         binding.buttonBackImage.setOnClickListener { navigateBack() }
         binding.buttonBackText.setOnClickListener { navigateBack() }
 
-        GlobalScope.launch(Dispatchers.Main) {
+        coroutineScope.launch(Dispatchers.Main) {
             val actors = getActors(requireContext())
             initAdapter(actors)
         }
```

Comment:
U should use IO or Default dispatcher for getActors()
And then use Main for actions related to UI.
To do this u need change Main to IO and use withContext(Main) for initAdapter

Reply:
This is IO if i'm not mistaken. 
On the initialisation line we can see _private var coroutineScope = CoroutineScope(**Dispatchers.IO** + SupervisorJob())_

Reply:
but `coroutineScope.launch(Dispatchers.Main)` means that u will proceed with Main dispatcher. 
u should use just launch() then and even better use withContext(IO) then even if u change ur coroutine this method will be safe

Reply:
Got it. Thanks. 
I've changed it to launch() since I had some problems with using withContext(IO)
Hope it's okay

Reply:
What kind of problem? U just need to move `val actors = getActors(requireContext())` into viewModel and then execute this new method from the viewMode and observe for actors via liveData.